### ISSUE TITLE
Apple picker, sometimes crashes on real data files

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#340]
For example `Falcon_2012_06_13-09_36_20_0.mrc` from the (Beta-galactosidase Falcon-II micrographs EMPIAR) dataset suggested in the getting started documentation; also where `sample.mrc` is drawn from.  I've narrowed it down to an empty set of centers, and I am trying to go through the code in that neighborhood a little.

Not finding particles is probably a reasonable thing to expect from time to time, so at the very least might want to handle a step in the process failing more gracefully.  In this case, with current code, if one was to run a whole directory, it would halt on such inputs and waste resources.

In case I am invoking it wrong, or pivot to another task:
```
python -m aspire apple --mrc_file   /scratch/work_files/10017/data/Falcon_2012_06_13-09_36_20_0.mrc --output_dir /scratch/work_files/10017/tmp
```

**Comment by janden** [comment]
> Not finding particles is probably a reasonable thing to expect from time to time

Eh it should be very rare. Unless that micrograph is trash we should be finding a couple of particles. That may be more of an algorithmic bug, though. At any rate, we shouldn't be crashing.

**Comment by garrettwrong** [comment]
Following up: Reviewed the code a little with Ayelet on 11/26.  We tried playing with some of the settings (max particle size) which had something like the expected effect, but did not actually identify or solve the problem.  Essentially, we are yielding way way too many particles as "too large", and so after throwing out "too small" and "too large" we have no particles leftover; then we crash. She is going to try to narrow it down and get back to me. 

Confirmed this entire dataset should work without any problems. However, it's not clear the whole dataset was ever run in Python, only the Matlab implementation.  I did go back in time (current master), and still had issues, so probably there is a small bug to find.  A few of the methods used may not be identical between python and matlab.

**Comment by janden** [comment]
Right. But do we know if we encountered this “no particles” problem in Matlab, even if it didn't crash?

**Comment by garrettwrong** [comment]
I am told, `No`. Ayelet was pretty confident about this. I have not confirmed independently yet. She offered to diagnose, so I was going to see how that goes...

**Comment by janden** [comment]
Ok sounds like a plan.

**Comment by garrettwrong** [comment]
Hrmm I didn't mean to close this.

Ayelet found that the result following binary_fill_holes is filling the entire image.  I get the impression from the docs that scipy's function may not be wquivalent to MATLAB's `imfill(.., 'holes')`.  I'm going to try and drop out the array prior to that call and run the one method in matlab.

**Comment by garrettwrong** [comment]
@chris-langfield , I wonder if your improvements to the box/coord sources might resolve this (once merged).

We might want to check that APPLE picker works for the data sets we have targeted, and what parameters might be reasonable.  For the parameters you would probably want to work with Ayelet in the future.
